### DISCUSSION TITLE
DirectML is getting updated again..

### DISCUSSION THREAD

**Discussion by dm3rd** [lshqqytiger/stable-diffusion-webui-directml#476] [General]
Seems microsoft has started updating directml again, I wonder if they're going to fix any of the major issues, like black squares when tiling large images or othe stuff?

**Comment by pw405** [comment]
If they could just get the VRAM usage/allocation a bit better, it would help!  Zluda has been amazing though, highly suggest.

**Reply by Hunter9886** [reply]
@pw405 does Zluda work with my pc if may ask?

RX 6700 XT (gpu)
Ryzen 5 1600 (cpu)

**Reply by lshqqytiger** [reply]
It is not officially supported, but you can run zluda with unofficially(user-) compiled BLAS libraries.
https://github.com/brknsoul/ROCmLibs

**Reply by Hunter9886** [reply]
@lshqqytiger 
if i do that and it works, should i use your version or SDnext? will it work? 

and also what commands should i type inside (COMMANDLINE_ARGS) while using Zluda??

**Reply by lshqqytiger** [reply]
Both work, add `--use-zluda`

**Reply by Hunter9886** [reply]
@lshqqytiger i waited a while for the first generation but it seems to work now normally with my GPU RX 6700 XT on SDnext, can i run pony/xl models with it or only 1.5 models like how it is with directml?

**Reply by lshqqytiger** [reply]
Try sdxl. It will be better than directml if zluda is working

**Reply by Hunter9886** [reply]
@lshqqytiger i just downloaded a pony model and after i added all my extenstions and embeddings back, i got this error:  

15:57:17-276966 INFO     Starting SD.Next
15:57:17-287976 INFO     Logger: file="C:\Users\Desktop\drx\automatic\sdnext.log" level=DEBUG size=65
                         mode=create
15:57:17-292130 INFO     Python version=3.10.6 platform=Windows bin="C:\Users\Desktop\drx\automatic\venv\Scripts\python.exe" venv="C:\Users\Desktop\drx\automatic\venv"
15:57:17-714028 INFO     Version: app=sd.next updated=2024-07-10 hash=2ec6e9ee branch=master
                         url=https://github.com/vladmandic/automatic/tree/master ui=main
15:57:18-230126 INFO     Platform: arch=AMD64 cpu=AMD64 Family 23 Model 1 Stepping 1, AuthenticAMD system=Windows
                         release=Windows-10-10.0.19045-SP0 python=3.10.6
15:57:18-238150 DEBUG    Setting environment tuning
15:57:18-241155 INFO     HF cache folder: C:\Users\.cache\huggingface\hub
15:57:18-244169 DEBUG    Torch allocator: "garbage_collection_threshold:0.80,max_split_size_mb:512"
15:57:18-246171 DEBUG    Torch overrides: cuda=False rocm=False ipex=False diml=False openvino=False
15:57:18-252165 DEBUG    Torch allowed: cuda=True rocm=True ipex=True diml=True openvino=True
15:57:18-274197 INFO     Python version=3.10.6 platform=Windows bin="C:\Users\Hozan\Desktop\CITRA
                         D\PW\drx\automatic\venv\Scripts\python.exe" venv="C:\Users\Hozan\Desktop\CITRA
                         D\PW\drx\automatic\venv"
15:57:18-287208 INFO     AMD ROCm toolkit detected
15:57:18-948579 DEBUG    ROCm agents detected: ['gfx1031']
15:57:18-951582 DEBUG    ROCm agent used by default: idx=0 gpu=gfx1031 arch=navi2x
15:57:18-984996 DEBUG    ROCm version detected: None
15:57:18-987999 WARNING  ZLUDA support: experimental
15:57:19-046052 INFO     Using ZLUDA in C:\Users\Desktop\drx\automatic\.zluda
15:57:19-422960 INFO     Verifying requirements
15:57:19-434971 INFO     Verifying packages
15:57:19-498891 DEBUG    Repository update time: Wed Jul 10 16:29:43 2024
15:57:19-502895 INFO     Startup: standard
15:57:19-504896 INFO     Verifying submodules
15:57:23-271670 DEBUG    Submodule: extensions-builtin/sd-extension-chainner / main
15:57:23-405793 DEBUG    Submodule: extensions-builtin/sd-extension-system-info / main
15:57:23-535911 DEBUG    Submodule: extensions-builtin/sd-webui-agent-scheduler / main
15:57:23-732089 DEBUG    Git detached head detected: folder="extensions-builtin/sdnext-modernui" reattach=main
15:57:23-736094 DEBUG    Submodule: extensions-builtin/sdnext-modernui / main
15:57:23-916257 DEBUG    Submodule: extensions-builtin/stable-diffusion-webui-rembg / master
15:57:24-048378 DEBUG    Submodule: modules/k-diffusion / master
15:57:24-239551 DEBUG    Git detached head detected: folder="wiki" reattach=master
15:57:24-242553 DEBUG    Submodule: wiki / master
15:57:24-319623 DEBUG    Register paths
15:57:24-564846 DEBUG    Installed packages: 241
15:57:24-567850 DEBUG    Extensions all: ['Lora', 'sd-extension-chainner', 'sd-extension-system-info',
                         'sd-webui-agent-scheduler', 'sdnext-modernui', 'stable-diffusion-webui-rembg']
15:57:25-297524 DEBUG    Running extension installer: C:\Users\Desktop\drx\automatic\extensions-builtin\sd-webui-agent-scheduler\install.py
15:57:26-409808 DEBUG    Running extension installer: C:\Users\Desktop\drx\automatic\extensions-builtin\stable-diffusion-webui-rembg\install.py
15:57:27-280588 DEBUG    Extensions all: ['adetailer', 'sd-civitai-browser-plus', 'sd-webui-3d-open-pose-editor',
                         'sd-webui-controlnet', 'sd-webui-reactor', 'Stylez', 'ultimate-upscale-for-automatic1111']
15:57:27-286595 DEBUG    Running extension installer: C:\Users\Desktop\drx\automatic\extensions\adetailer\install.py
15:57:28-189415 DEBUG    Running extension installer: C:\Users\Desktop\drx\automatic\extensions\sd-civitai-browser-plus\install.py
15:57:41-167625 DEBUG    Running extension installer: C:\Users\Desktop\drx\automatic\extensions\sd-webui-3d-open-pose-editor\install.py
15:57:41-817418 DEBUG    Running extension installer: C:\Users\Desktop\drx\automatic\extensions\sd-webui-controlnet\install.py
15:57:42-704668 DEBUG    Running extension installer: C:\Users\Desktop\drx\automatic\extensions\sd-webui-reactor\install.py
15:57:47-605263 INFO     Extensions enabled: ['Lora', 'sd-extension-chainner', 'sd-extension-system-info',
                         'sd-webui-agent-scheduler', 'sdnext-modernui', 'stable-diffusion-webui-rembg', 'adetailer',
                         'sd-civitai-browser-plus', 'sd-webui-3d-open-pose-editor', 'sd-webui-controlnet',
                         'sd-webui-reactor', 'Stylez', 'ultimate-upscale-for-automatic1111']
15:57:47-614271 INFO     Verifying requirements
15:57:47-617274 DEBUG    Setup complete without errors: 1720879068
15:57:47-630286 DEBUG    Extension preload: {'extensions-builtin': 0.0, 'extensions': 0.0}
15:57:47-634289 DEBUG    Starting module: <module 'webui' from 'C:\\Users\\Desktop\\CITRA
                         D\\PW\\drx\\automatic\\webui.py'>
15:57:47-640295 INFO     Command line args: ['--use-zluda', '--debug', '--autolaunch'] autolaunch=True use_zluda=True
                         debug=True
15:57:47-648302 DEBUG    Env flags: []
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ C:\Users\Desktop\drx\automatic\launch.py:265 in <module>                                           │
│                                                                                                                     │
│   264 if __name__ == "__main__":                                                                                    │
│ > 265     main()                                                                                                    │
│   266                                                                                                               │
│                                                                                                                     │
│ C:\Users\Desktop\drx\automatic\launch.py:242 in main                                               │
│                                                                                                                     │
│   241                                                                                                               │
│ > 242     uv, instance = start_server(immediate=True, server=None)                                                  │
│   243     while True:                                                                                               │
│                                                                                                                     │
│ C:\Users\Desktop\drx\automatic\launch.py:168 in start_server                                       │
│                                                                                                                     │
│   167     get_custom_args()                                                                                         │
│ > 168     module_spec.loader.exec_module(server)                                                                    │
│   169     uvicorn = None                                                                                            │
│ in exec_module:883                                                                                                  │
│ in _call_with_frames_removed:241                                                                                    │
│                                                                                                                     │
│ C:\Users\Desktop\drx\automatic\webui.py:11 in <module>                                             │
│                                                                                                                     │
│    10 from threading import Thread                                                                                  │
│ >  11 import modules.loader                                                                                         │
│    12 import torch # pylint: disable=wrong-import-order                                                             │
│                                                                                                                     │
│ C:\Users\Desktop\drx\automatic\modules\loader.py:38 in <module>                                    │
│                                                                                                                     │
│   37                                                                                                                │
│ > 38 import onnxruntime # pylint: disable=W0611,C0411                                                               │
│   39 onnxruntime.set_default_logger_severity(3)                                                                     │
│                                                                                                                     │
│ C:\Users\Desktop\drx\automatic\venv\lib\site-packages\onnxruntime\__init__.py:57 in <module>       │
│                                                                                                                     │
│   56 if import_capi_exception:                                                                                      │
│ > 57     raise import_capi_exception                                                                                │
│   58                                                                                                                │
│                                                                                                                     │
│ C:\Users\\Desktop\drx\automatic\venv\lib\site-packages\onnxruntime\__init__.py:23 in <module>       │
│                                                                                                                     │
│   22 try:                                                                                                           │
│ > 23     from onnxruntime.capi._pybind_state import ExecutionMode  # noqa: F401                                     │
│   24     from onnxruntime.capi._pybind_state import ExecutionOrder  # noqa: F401                                    │
│                                                                                                                     │
│ C:\Users\Desktop\drx\automatic\venv\lib\site-packages\onnxruntime\capi\_pybind_state.py:32 in <mod │
│                                                                                                                     │
│   31                                                                                                                │
│ > 32 from .onnxruntime_pybind11_state import *  # noqa                                                              │
│   33                                                                                                                │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
ImportError: DLL load failed while importing onnxruntime_pybind11_state: A dynamic link library (DLL) initialization routine failed.
Press any key to continue . . .

**Reply by lshqqytiger** [reply]
have you launched sd.next with directml?

**Reply by Hunter9886** [reply]
@lshqqytiger no, with:

call conda activate sdnext 
webui.bat --use-zluda --debug --autolaunch

**Reply by lshqqytiger** [reply]
Don't run webui.bat with anaconda.
Remove venv folder and launch webui directly.
```
conda activate sdnext
python launch.py --use-zluda --debug --autolaunch
```
anaconda has issue with dll files that are built with Visual C 17.10.x.
upgrade Visual C Runtime in anaconda environment by running this command.
```
conda install conda-forge::vs2015_runtime
```

**Reply by Hunter9886** [reply]
@lshqqytiger  

conda activate sdnext
python launch.py --use-zluda --debug --autolaunch (this doesnt activate)

these commands only puts me in the virtual environment of sdnext, it doesnt open and create a new env folder or anything, so i just opened the normal webui.bat file in the sdnext directory and it gave me this error : 

Using python: "D:\apps\mini\python.exe"
Creating VENV: C:\Users\Desktop\drx\automatic\venv
Using VENV: C:\Users\Desktop\drx\automatic\venv
22:00:02-234117 INFO     Starting SD.Next
22:00:02-245350 INFO     Logger: file="C:\Users\Desktop\drx\automatic\sdnext.log" level=INFO size=6427
                         mode=append
22:00:02-250355 INFO     Python version=3.12.4 platform=Windows bin="C:\Users\\Desktop\drx\automatic\venv\Scripts\python.exe" venv="C:\Users\Desktop\drx\automatic\venv"
22:00:02-256360 ERROR    Incompatible Python version: 3.12.4 required 3.12.3 or lower
Press any key to continue . . .

Edit: i ran your commands manually through cmd and it worked, but i want to have a shortcut with the commands you gave me cuz it activates the virtual environment but then doesnt run:  python launch.py --use-zluda --debug --autolaunch 

i wanna run SD without having to activate/entering the VE always, how can i create a shortcut with your commands??

**Reply by lshqqytiger** [reply]
create a batch file, or just use local python <3.12. (webui.bat will automatically create and activate virtual environment)

**Reply by Hunter9886** [reply]
@lshqqytiger when creating a batch file it just opens the virutal enviorment and doesn't automatically open Sdnext, wdym with local python? should i upgrade pyhton from 3.10.6 to newest version and it will be fine with Sdnext/zluda?

because right now i only can open it manually with activating the VE, then using your command : python launch.py --use-zluda --debug --autolaunch

**Reply by lshqqytiger** [reply]
- Using batch file
Create a batch file in SD.Next folder
```
D:\miniconda3\Scripts\activate.bat
conda activate sdnext
python launch.py --use-zluda --debug --autolaunch
```

- Using python venv
1. Uninstall python 3.12.4. (you have python 3.12.4)
2. Install python 3.11.x or 3.10.x.
3. Just run webui.bat.

**Reply by Hunter9886** [reply]
@lshqqytiger 
i Uinstalled python 3.12.4 miniconda, then i got this: 

DirectML installation is detected. Skipping HIP SDK check.
15:28:40-056678 WARNING  ZLUDA failed to initialize: no HIP SDK found
then it installed a lot of packages through pip which i should've already downloaded, is something wrong?

https://github.com/vladmandic/automatic/discussions/2940#discussion-631147

i think i need to uninstall some things regarding directml because i was using your version?

**Reply by lshqqytiger** [reply]
Just remove venv folder and launch webui again.

**Reply by Hunter9886** [reply]
@lshqqytiger 
this all became confusing with me so ill just reinstall everything according to https://youtu.be/DSfiLpykd_c?si=6bHQHhAI0vEs_7-l

![Python programms](https://github.com/user-attachments/assets/2ec2ba93-906a-40a3-908b-b4c0760c44b8)
with regarding to those 2 python versions, should i keep both or just one version as you said? 
i have Python 3.10.6 and miniconda exactly like in the attached files. 

zluda worked when i had miniconda3 python 3.12.4(i just had to activate the VE everytime as i said), i uninstalled it and then these new problems came. what should i do? should i reinstall [vladmandic/automatic) everything with 1 version of python or both because in his video i need miniconda3 but according to you i have uninstall it because it has issues with dll files as u said? 


conda activate sdnext
python launch.py --use-zluda --debug --autolaunch = these 2 codes work for him but not for me?? mine just stops after opening the Virtual E. 


불편을 끼쳐드려 정말 죄송합니다.

**Reply by lshqqytiger** [reply]
Having miniconda and Python is not a problem, but you should understand which version of Python you are using.

You don't need to use miniconda at all. It is useful for developers, but it is overhead for those who do not need multiple versions of Python.

It should not be a problem if you activate and run webui using only one virtual environment regardless of whether it is miniconda environment or python-venv, although you have miniconda and Python both.

**Reply by Hunter9886** [reply]
@lshqqytiger but according to his video i need miniconda, conda create --name sdnext python=3.10.6 works only if i have miniconda, how can install the VE of sdnext without conda using just python 3.10.6?

im very sorry im just so unexperienced on this. i just wanna run SD zluda with my gpu.

**Reply by lshqqytiger** [reply]
Python itself has virtual environment system. (python-venv in Linux)
webui.bat automatically creates a virtual environment for running webui. (venv folder)

**Reply by Hunter9886** [reply]
@lshqqytiger the first installation with miniconda, i didnt have a venv folder within sdnext even after it worked. (it said that it has created a venv but there was no folder after i checked it) all good tho, i thank u for all the help as always sorry again i will try to reinstall everything again and see what happens.

**Reply by lshqqytiger** [reply]
miniconda does not create venv folder, instead, it manages virtual environment directly. Python creates venv folder.
webui.bat is designed for Python, not for miniconda.

**Reply by Hunter9886** [reply]
@lshqqytiger i've managed to reinstall everything and i do manually open a created batch file with :

conda activate sdnext
python launch.py --use-zluda --debug --autolaunch

i only get this thing in the end, while the UI is already open and working. 


ERROR    API error: GET: http://127.0.0.1:7860/file=placeholder {'error': 'HTTPException', 'code': 404,
                         'detail': 'File not found: placeholder.', 'body': '', 'errors': '404: File not found:

i can generate txt2img, it works. i just have this error above is it a problem?

**Comment by ride5k** [comment]
has anyone any experience post-update?

agreed that zluda has been invigorating.

**Reply by Hunter9886** [reply]
@ride5k and everyone who sees this reply:

i wanted to say that if u guys have an amd gpu (mine is 6700 xt) and wanna find the right settings for sdnext to work with zluda after installing everything from [vladmandic] or a Tutorial, join
sdnext discord channel>support forums> (question For 6700 xt (Ratio of images generated) is the name of my Post 


there u can fine the right settings and generate up to 768x768 or 768x1024 or vice versa with an 6700xt with those settings that are mentioned and have already been tested by me. 

just wanted to mention it in case it helps someone who has a 6700 xt amd gpu and wants zluda to work with it.

peace!
